### PR TITLE
Fix animating vectors and quaternions that are not position, rotation or scale.

### PR DESCRIPTION
**Pull request by SG-Oleksandr**

### CODE REVIEW

**Comment by SG-Oleksandr** [comment]
Encountered a bug where it would print unnecessary warnings when sampler's interpolation is cubic spline, fixing...

**Comment by SG-Oleksandr** [comment]
> Encountered a bug where it would print unnecessary warnings when sampler's interpolation is cubic spline, fixing...

Fixed.

File: src/playcanvas-gltf.js [hunk 1]
```diff
@@ -141,19 +141,44 @@ Object.assign(window, function () {
...
+
+                var valuesContainTangents = sampler.interpolation === "CUBICSPLINE";
+
+                if (valuesContainTangents)
+                    numCurves /= 3;
+
+                switch (Math.round(numCurves)) {
+                    case 1: keyType = AnimationKeyableType.NUM; break;
+                    case 3: keyType = AnimationKeyableType.VEC; break;
+                    case 4: keyType = AnimationKeyableType.QUAT; break;
```
Comment (thomasbiang) [inline src/playcanvas-gltf.js]: This part affects default keyType, based on numCurves.  If path has specific name, like "translation", keyType will still be assigned by function ensureKeyType as before.

File: src/playcanvas-gltf.js [hunk 2]
```diff
@@ -141,19 +141,44 @@ Object.assign(window, function () {
                 }
             } else {
                 // translation, rotation or scale
-                keyType = AnimationKeyableType.NUM;
+
```
Comment (thomasbiang) [inline src/playcanvas-gltf.js]: For path is not 'weights', one channel only has one corresponding AnimationCurve. Key on that AnimationCurve could be a single number (NUM), or a 3d vector (VEC) or a 4d quaternion (QUAT). But there is only one curve, animating 1d, 3d or 4d data together. 

So here the logic seems is depending on the DOF (degree of freedom) of animation data, decide whether the curve should have either NUM, VEC or QUAT key.  The name numCurves is a bit confusing. The logic is good.

File: src/playcanvas-gltf.js [hunk 3]
```diff
@@ -141,19 +141,44 @@ Object.assign(window, function () {
...
+                    if (keyType !== expectedKeyType) {
+                        console.warn(
+                            'Unexpected key type: ' +
+                            '"' + keyType + '" ' +
+                            '(expected "' + expectedKeyType + '")'
+                        );
+                        keyType = expectedKeyType;
+                    }
+                }
+
```
[the remark this answers is not in the document]
Comment (SG-Oleksandr) [reply src/playcanvas-gltf.js]: There was one when I first forgot to account for cubic spline sampler interpolation, but after I fixed that in the last commit of this PR I've seen no warnings showing in my project (~30 different animations for a skeletal bipedal character and ~5 general purpose animations)

**Comment by willeastcott** [comment]
Thanks for the PR! 😄

File: src/playcanvas-gltf.js (hunk 2 above)
Comment (SG-Oleksandr) [reply src/playcanvas-gltf.js]: renamed to `valueSize` in  https://github.com/SG-Oleksandr/playcanvas-gltf/commit/ec6063e6f4227b99f58cc927251ad45cc5cadb4f

**Pull request merged**